code/__DEFINES/multi_z.dm

```diff
@@ -9,7 +9,30 @@
 #define Z_LEVEL_DOWN 2
 #define LARGEST_Z_LEVEL_INDEX Z_LEVEL_DOWN
 
-/// Attempt to get the turf below the provided one according to Z traits
-#define GET_TURF_BELOW(turf) ((!(turf) || !length(SSmapping.multiz_levels) || !SSmapping.multiz_levels[(turf).z][Z_LEVEL_DOWN]) ? null : get_step((turf), DOWN))
-/// Attempt to get the turf above the provided one according to Z traits
-#define GET_TURF_ABOVE(turf) ((!(turf) || !length(SSmapping.multiz_levels) || !SSmapping.multiz_levels[(turf).z][Z_LEVEL_UP]) ? null : get_step((turf), UP))
+/// The below procs require you to either directly pass in a turf, or assign to a turf typed variable
+/// The unsafe access to above needs to be null checked otherwise the compiler gets confused with the ternary
+/// expression.
+
+/// Attempt to get the turf below the provided one according to either
+/// the map config defined z-traits, or the assigned below turf.
+#define GET_TURF_BELOW(turf) \
+( \
+	!(turf) ? null : ( \
+		(turf).below || ( \
+			(turf).set_below(MAPPING_TURF_BELOW(turf)) \
+		) \
+	) \
+)
+#define MAPPING_TURF_BELOW(turf) ((!(turf) || !length(SSmapping.multiz_levels) || !SSmapping.multiz_levels[(turf).z][Z_LEVEL_DOWN]) ? null : get_step((turf), DOWN))
+
+/// Attempt to get the turf above the provided one according to either
+/// the map config defined z-traits, or the assigned above turf
+#define GET_TURF_ABOVE(turf) \
+( \
+	!(turf) ? null : ( \
+		(turf).above || ( \
+			(turf).set_above(MAPPING_TURF_ABOVE(turf)) \
+		) \
+	) \
+)
```

Comment:
TODO: We need a flag on turfs in order to identify when there is no turf below so that we don't keep going into set_above calls over and over due to not being able to cache a below value.

Reply:
is this resolved or not important currently?